Release notes 2 2 9 update (#4785)
magento/devdocs: guides/v2.2/release-notes/ReleaseNotes2.2.9EE.md

---

> *Patch code and release notes published on June 25 2019.*

Author: ```suggestion
*Patch code and release notes published on June 25, 2019.*
```

### Installation, setup, and deployment

<!-- MAGETWO-76424 -->

> * Magento no longer displays an extraneous blank option in the country drop-down menu on the Country Options page for store configuration settings.

Author: ```suggestion
* Magento no longer displays an extraneous blank option in the country drop-down menu on the Country Options page for store configuration settings (**Stores** > Settings > **Configuration** > **General** > **Country Options**).
```

### Installation, setup, and deployment

> <!-- MAGETWO-98860 -->

Author: ```suggestion
```

> * Magento no longer throws an error when executing `setup:static-content:deploy` in parallel mode if theme or locale deployment takes more than 400 seconds. Previously, Magento threw the following error under these conditions: `2436; Status: 0`.

Author: ```suggestion
```

<!-- MAGETWO-76424 -->

> * Magento no longer displays an extraneous blank option in the country drop-down menu.

Author: ```suggestion
* Magento no longer displays an extraneous blank option in the country drop-down menu on the Country Options page for store configuration settings.
```

> * Magento no longer displays an extraneous blank option in the country drop-down menu.

Author: ```suggestion
* Magento no longer displays an extraneous blank option in the country drop-down menu on the Country Options page for store configuration settings.
```

> * Magento no longer displays an extraneous blank option in the country drop-down menu.

Author: ```suggestion
* In the **Country Options** section in the general store configuration settings ( **Stores**> Settings > **Configuration** > **General**), Magento no longer displays an extraneous blank option in the *Country* drop-down menu. 
```

<!-- MAGETWO-98553 -->
* The `catalogpermissions_category` indexer in Update On Schedule mode now correctly picks up on the structure of a shared catalog.

<!-- MAGETWO-98678 -->
* You can now add backordered products to the cart when backorders are enabled.

<!-- MAGETWO-95281 -->

> * Quotes now remain associated with the customer who generated it  when you move the customer from one company to another. Previously, a quote did not appear in the customer’s account on the storefront after the move. 

Author: ```suggestion
* Quotes now remain associated with the customer who generated the quote when you move the customer from one company to another. Previously, a quote did not appear in the customer’s account on the storefront after the move. 
```

* Magento no longer increments stock for products for which stock managing has been disabled. Previously, Magento increased the product quantity count when an order failed if **Manage Stock** was disabled. *Fix submitted by [Amol Chaudhari](https://github.com/amol2jcommerce) in pull request [20644](https://github.com/magento/magento2/pull/20644)*. [GitHub-19482](https://github.com/magento/magento2/issues/19482)

<!-- ENGCOM-4514 -->
* * We have replaced the incorrect proxy `resourceStock` argument for the `\Magento\CatalogInventory\Observer\UpdateItemsStockUponConfigChangeObserver` in `di.xml`. (Specifically, `<argument name="resourceStock" xsi:type="object">Magento\CatalogInventory\Model\ResourceModel\Stock\Proxy</argument>`

>  with the following argument:

Author: ```suggestion
 has been replaced with the following argument:
```

  `<argument name="resourceStockItem" xsi:type="object">Magento\CatalogInventory\Model\ResourceModel\Stock\Item\Proxy</argument>`. ) *Fix submitted by [Vitaliy](https://github.com/VitaliyBoyko) in pull request [21793](https://github.com/magento/magento2/pull/21793)*. [GitHub-167](https://github.com/magento/magento2/issues/167)

<!-- MAGETWO-98794 -->
* You can now add a product to an order by using the **Add products by SKU** button. Previously, when you tried to add a product using this method, Magento displayed the following message: `The SKU was not found in the catalog.

<!-- ENGCOM-4823 -->
* Magento now increments product quantity correctly when you add products to your cart first as a guest user, and then logged in. Previously, Magento added items separately instead. *Fix submitted by [Amol Chaudhari](https://github.com/amol2jcommerce) in pull request [22378](https://github.com/magento/magento2/pull/22378)*. [GitHub-21375](https://github.com/magento/magento2/issues/21375)

> <!-- ENGCOM-4477 -->

Author: ```suggestion
```

> * When you configure a price rule for configurable products with swatches, Magento now shows the special price for products that match the price rule. Previously, Magento displayed both the old price and the special price for the matching configurable products. *Fix submitted by [Pratik Oza](https://github.com/mage2pratik) in pull request [21695](https://github.com/magento/magento2/pull/21695)*. [GitHub-19276](https://github.com/magento/magento2/issues/19276)

Author: ```suggestion
```

  
<!-- ENGCOM-4253 -->
* **Meta Keywords** and **Meta Description** are now defined as textarea throughout product forms. *Fix submitted by [Amol Chaudhari](https://github.com/amol2jcommerce) in pull request [21199](https://github.com/magento/magento2/pull/21199)*. [GitHub-20555](https://github.com/magento/magento2/issues/20555) 

### Catalog rule

<!-- MAGETWO-98679 -->

> * Catalog rules are applied as expected after a customer uses Visual Merchandiser  to drag and drop products in a category. Previously, discounts that were applied to products through a catalog rule were dropped after the product was dragged and dropped.

Author: ```suggestion
* Catalog rules are applied as expected after a customer uses Visual Merchandiser to drag and drop products in a category. Previously, discounts that were applied to products through a catalog rule were dropped after the product was dragged and dropped.
```

<!-- ENGCOM-4518 -->
* Magento no longer throws an exception when the `fixed-Quantity_and_stock_status` attribute has **Change Visible on Catalog Pages on Storefront** enabled. *Fix submitted by [Amol Chaudhari](https://github.com/amol2jcommerce) in pull request [21802](https://github.com/magento/magento2/pull/21802)*. [GitHub-13612](https://github.com/magento/magento2/issues/13612)

<!-- ENGCOM-4787 -->
* The `\Magento\Eav\Model\Entity\Collection\AbstractCollection::importFromArray()` method now returns a usable collection. Previously, the `_isCollectionLoaded` property was false, and every interaction threw an exception. *Fix submitted by [Lorenzo Stramaccia](https://github.com/slackerzz) in pull request [22422](https://github.com/magento/magento2/pull/22422)*. [GitHub-21868](https://github.com/magento/magento2/issues/21868)

### Email

<!-- ENGCOM-4305 -->
* Corrected problems with disabling and enabling order-related emails. *Fix submitted by [Serhiy Zhovnir](https://github.com/serhiyzhovnir) in pull request [20954](https://github.com/magento/magento2/pull/20954)*. [GitHub-18698](https://github.com/magento/magento2/issues/18698)

<!-- ENGCOM-4824 -->

> * Magento no longer sends via asynchronous email sending any sales-related emails that were created when email sending was disabled once email sending is enabled. *Fix submitted by [Serhiy Zhovnir](https://github.com/serhiyzhovnir) in pull request [22108](https://github.com/magento/magento2/pull/22108)*. [GitHub-21786](https://github.com/magento/magento2/issues/21786)

Reviewer: how about this? marginally better but clearer: Magento no longer asynchronously sends sales-related emails that were created when email sending was disabled.

<!-- ENGCOM-4272 -->
* You can now change customer groups when creating a new customer during order creation on the Admin. *Fix submitted by [Amol Chaudhari](https://github.com/amol2jcommerce) in pull request [21239](https://github.com/magento/magento2/pull/21239)*. [GitHub-7974](https://github.com/magento/magento2/issues/7974)

<!-- MAGETWO-98000 -->

> * The store switcher now redirects users to the correct store. 

Author: ```suggestion
* PayPal Express now creates a gift card as expected when payment for the card is captured. Previously, Magento did not create a gift card under these circumstances, even when payment action was set to **Authorize and Capture** and **Generate Gift Card Account when Order Item is** was set  to *Invoiced*.
```

<!-- ENGCOM-4272 -->
* You can now change customer groups when creating a new customer during order creation on the Admin. *Fix submitted by [Amol Chaudhari](https://github.com/amol2jcommerce) in pull request [21239](https://github.com/magento/magento2/pull/21239)*. [GitHub-7974](https://github.com/magento/magento2/issues/7974)

<!-- MAGETWO-98000 -->

> * The store switcher now redirects users to the correct store. 

Author: Incorrect info for this release. Added correct information.

<!-- ENGCOM-4702 -->
* Widget parameters can now contain multidimensional arrays. *Fix submitted by [Stanislav Ilnytskyi](https://github.com/ilnytskyi) in pull request [22214](https://github.com/magento/magento2/pull/22214)*. [GitHub-19909](https://github.com/magento/magento2/issues/19909)

### Layered navigation

<!-- ENGCOM-4803 -->
* Setting **price navigation step calculation** for layered navigation to **Automatic (equalize product counts)** now works as expected. Previously, results were not in the equals range, but omitted products. *Fix submitted by [Nazar Klovanych](https://github.com/Nazar65) in pull request [22453](https://github.com/magento/magento2/pull/22453)*. [GitHub-6715](https://github.com/magento/magento2/issues/6715), [GitHub-21960](https://github.com/magento/magento2/issues/21960)
  
### Logging

<!-- MAGETWO-97595 -->
* Editing a theme now creates an entry in the Admin action log as expected. 

> <!-- ENGCOM-4834 -->

Author: ```suggestion
```

> * The downloads report table (**Admin** > **Reports** > **Downloads**) now displays an accurate count of all downloadable products and the number of times they have been downloaded. *Fix submitted by [Shikha Mishra](https://github.com/shikhamis11) in pull request [22523](https://github.com/magento/magento2/pull/22523)*. [GitHub-22223](https://github.com/magento/magento2/issues/22223)

Author: ```suggestion
```

<!-- ENGCOM-4366 -->
* You can now programmmatically upload an image for the customer attribute. Previously, Magento threw the following error: `error: Base64 is not defined`. *Fix submitted by [Nazar Klovanych](https://github.com/Nazar65) in pull request [21437](https://github.com/magento/magento2/pull/21437)*. [GitHub-19983](https://github.com/magento/magento2/issues/19983)

<!-- ENGCOM-4444 -->
* Corrected tabbing issue on the product page. *Fix submitted by [Amol Chaudhari](https://github.com/amol2jcommerce) in pull request [21588](https://github.com/magento/magento2/pull/21588)*. [GitHub-21077](https://github.com/magento/magento2/issues/21077)

<!-- ENGCOM-4361 -->
* If you retrieve an order, and then use the `getShippingMethod` as object function to retrieve the shipping method, Magento now returns `null` if no shipping method has been defined. Previously, this function returned an undefined index error if a shipping method was not available. *Fix submitted by [Mahesh Singh](https://github.com/maheshWebkul721) in pull request [20866](https://github.com/magento/magento2/pull/20866)*. [GitHub-20380](https://github.com/magento/magento2/issues/20380)

<!-- ENGCOM-4820 -->

> * Magento now cancels previous scrolling actions as expected when you click Add to cart on a product page. Previously, Magento scrolled back to the `qty` input box the same number of times as you clicked Add to cart. *Fix submitted by [Amol Chaudhari](https://github.com/amol2jcommerce) in pull request [22358](https://github.com/magento/magento2/pull/22358)*. [GitHub-21715](https://github.com/magento/magento2/issues/21715)

Reviewer 2: ```suggestion
* Magento now cancels previous scrolling actions as expected when you click **Add to cart** on a product page. Previously, Magento scrolled back to the `qty` input box the same number of times as you clicked **Add to cart**. *Fix submitted by [Amol Chaudhari](https://github.com/amol2jcommerce) in pull request [22358](https://github.com/magento/magento2/pull/22358)*. [GitHub-21715](https://github.com/magento/magento2/issues/21715)
```

<!-- ENGCOM-4232 -->
* Magento can now validate Greek VAT numbers. *Fix submitted by [Amol Chaudhari](https://github.com/amol2jcommerce) in pull request [21169](https://github.com/magento/magento2/pull/21169)*. [GitHub-6960](https://github.com/magento/magento2/issues/6960)

### Web API framework

<!-- MAGETWO-73978 -->

> * Magento now throws an error as expected when you send an APU request that specifies `pageSize` and `currentPage` values that exceed the number of products in the catalog you’re querying. Previously, when you queried a catalog and specified values that exceeded catalog size, the API would ignores the specified `currentPage` value, and returned the same results as the highest valid `currentPage`. 

Reviewer 2: ```suggestion
* Magento now throws an error as expected when you send an API request that specifies `pageSize` and `currentPage` values that exceed the number of products in the catalog you’re querying. Previously, when you queried a catalog and specified values that exceeded catalog size, the API would ignores the specified `currentPage` value, and returned the same results as the highest valid `currentPage`. 
```

<!-- ENGCOM-4792 -->

> * You can now use REST in scope `all` to save an existing category that does not have a name attribute. Previously, Magento threw this exception, `Could not save category with message. The "Name" attribute value is empty. Set the attribute and try again`. *Fix submitted by [Saphal Jha](https://github.com/saphaljha) in pull request [22440](https://github.com/magento/magento2/pull/22440)*. [GitHub-22309](https://github.com/magento/magento2/issues/22309)

Reviewer 2: ```suggestion
* You can now use the `PUT /rest/all/V1/categories/:id` call to save an existing category that does not have a name attribute. Previously, Magento threw this exception, `Could not save category with message. The "Name" attribute value is empty. Set the attribute and try again`. *Fix submitted by [Saphal Jha](https://github.com/saphaljha) in pull request [22440](https://github.com/magento/magento2/pull/22440)*. [GitHub-22309](https://github.com/magento/magento2/issues/22309)
```
